coding-articles/natachi-article-introduction-smart-contract-auditing/Article.md

```diff
@@ -0,0 +1,494 @@
+    
+3. Running fast tools - Tools like Slither are used to quickly identify potential vulnerabilities and security issues in the code.
+    
+4. Manual analysis - After running the tests and using tools, a manual analysis of the code is performed to identify any issues that may have been missed by the automated processes.
+    
+5. Discussion - Any identified issues are discussed to ensure a thorough understanding of the problem and to determine the best course of action.
+    
+6. Writing the report - Finally, a comprehensive report is written that documents all the identified issues, including recommendations for fixing any problems. The report can be used as a reference for future updates or audits of the smart contract.
+    
+
+Okayyy! Let's jump into auditing a code base!
+
+Remember, we'll be working on the project using Foundry and Slither. [Click here](https://github.com/natachigram/audit-practice.git) to access the project's github repository.
+
+1. **First, open Vscode. Then, make a new folder and clone the** [**github repo**](https://github.com/natachigram/audit-practice.git) **for this task. After that, go to your terminal and type "forge install".**
+    
+    ```plaintext
+    $ https://github.com/natachigram/audit-practice.git
+    $ cd audit-practice
+    $ forge install
+    ```
+    
+    In the <mark>audit/src/ </mark> folder, there are three contracts written in Solidity. We will be using these contracts for this tutorial. Also, remember to look at the <mark>audit/test/</mark> folder, as we will discuss it later.
+    
+2. You can do static analysis with Slither in two ways. You can either use it in the terminal or as an [extension](https://marketplace.visualstudio.com/items?itemName=trailofbits.slither-vscode) in vscode. We'll explain both options so it's easy to understand.
+    
+    **On your terminal run** `slither .`
+    
+    ```plaintext
+    slither .
+    ```
+    
+    You'll see some results in three groups: high (green), medium (yellow), and low (green) vulnerabilities. You can achieve the same outcome by using the slither extension in vscode. **Just click on the slither icon and then run it.**
+    
+
+Slither found two high level risks:
+
+The ReentrancyExample.sol contract has a <mark>reentrancy issue.</mark>
+
+The OverUnderVul.sol contract uses a <mark>weak PRNG</mark>.
+
+<img src="images/vul.png" style="width:100%;height:350px"/>       
+
+If you examine closely, you'll see that Slither missed a major problem in the OverUnderVul.sol contract and ReplayVul.sol. That's why it's important to also manually go through the code one line at a time.
+
+1. **Manual Code Review:**
+    
+    we will going over the **OverUnderVul.sol which is a Lottery contract first.**
+    
+    ```solidity
+    // SPDX-License-Identifier: MIT
+    pragma solidity ^0.8.0;
+    
+    contract Lottery {
+        address public owner;
+        uint public jackpot;
```

Comment:
uint = uint256, no?

So this is already the largest possible value.

Maybe change this to _`uint32 public jackpot`_ so that you can make your point about jackpot overflow more clearly.

Also pls let me know if I'm making a conceptual mistake in interpreting this.